src/instructlab/model/phased_training.py

```diff
@@ -0,0 +1,196 @@
+    score: float
+
+    @pydantic.field_serializer("checkpoint", "ended_at_utc")
+    def call_str_constructor(self, val: typing.Any) -> str:
+        return str(val)
+
+
+class EvalPhaseModel(pydantic.BaseModel):
+    """Stores info about evaluation phase"""
+
+    started_at_utc: datetime.datetime = AutoDatetimeField
+    ended_at_utc: datetime.datetime | None = None
+    checkpoints: list[pydantic.DirectoryPath]
+    finished_checkpoints: list[pydantic.DirectoryPath] = []
+    results: list[EvalResult] = []
+    best_checkpoint: EvalResult | None = None
+
+    @pydantic.field_serializer(
+        "checkpoints",
+        "finished_checkpoints",
+    )
+    def pathlibPath_list_to_str(self, paths: list[pathlib.Path]) -> list[str]:
+        return [str(path) for path in paths]
+
+    @pydantic.field_serializer("ended_at_utc")
+    def serialize_optional_datetime(self, val: datetime.datetime | None):
+        if val:
+            return str(val)
+        return val
+
+
+class TrainPhaseModel(pydantic.BaseModel):
+    """Stores info about training phase. Doesn't store training args because user can get those elsewhere, e.g. config."""
+
+    started_at_utc: datetime.datetime = AutoDatetimeField
+    ended_at_utc: datetime.datetime | None = None
+    checkpoints: pydantic.DirectoryPath
+    # TODO: might want training args here, but can't currently do it
+    # without side-effects because model instance is changed inside of _training_phase
+
+    @pydantic.field_serializer("checkpoints")
+    def call_str_constructor(self, val: typing.Any) -> str:
+        return str(val)
+
+    @pydantic.field_serializer("ended_at_utc")
+    def serialize_optional_datetime(self, val: datetime.datetime | None):
+        if val:
+            return str(val)
+        return val
+
+
+class JournalModel(pydantic.BaseModel):
+    run_id: uuid.UUID = pydantic.Field(default_factory=uuid.uuid4)
+    started_at_utc: datetime.datetime = AutoDatetimeField
+    ended_at_utc: datetime.datetime | None = None
+    current_phase: TrainingPhases = TrainingPhases.TRAIN1
+    train_1: TrainPhaseModel | None = None
+    eval_1: EvalPhaseModel | None = None
+    train_2: TrainPhaseModel | None = None
+    eval_2: EvalPhaseModel | None = None
+    final_output: EvalResult | None = None
+
+    @pydantic.field_serializer("current_phase")
+    def enum_to_value(self, enum_val: TrainingPhases) -> str:
+        return str(enum_val.value)
+
+    @pydantic.field_serializer("run_id")
+    def object_to_str(self, value: typing.Any) -> str:
+        return str(value)
+
+    @pydantic.field_serializer("ended_at_utc")
+    def serialize_optional_datetime(self, val: datetime.datetime | None):
+        if val:
+            return str(val)
+        return val
+
+
+class TrainingJournal:
+    def __init__(self, journalfile: pathlib.Path):
+        self.journalfile = journalfile
+
+        if not journalfile.exists():
+            raise ValueError(f"Path to journal must exist. {str(journalfile)}")
```

Comment:
Why are we erroring if the Journal doesn't exist? This doesn't seem correct

Reply:
fixed